ext/reflection/php_reflection.c

```diff
@@ -2953,27 +2953,49 @@ ZEND_METHOD(reflection_method, __construct)
 	char *name_str, *tmp;
 	size_t name_len, tmp_len;
 	zval ztmp;
+	const char *space;
 
-	if (zend_parse_parameters_ex(ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS(), "zs", &classname, &name_str, &name_len) == FAILURE) {
-		if (zend_parse_parameters_throw(ZEND_NUM_ARGS(), "s", &name_str, &name_len) == FAILURE) {
-			return;
-		}
+	if (zend_parse_parameters_throw(ZEND_NUM_ARGS(), "z|s!", &classname, &name_str, &name_len) == FAILURE) {
+		return;
+	}
 
-		if ((tmp = strstr(name_str, "::")) == NULL) {
-			zend_throw_exception_ex(reflection_exception_ptr, 0,
-				"Invalid method name %s", name_str);
-			return;
-		}
-		classname = &ztmp;
-		tmp_len = tmp - name_str;
-		ZVAL_STRINGL(classname, name_str, tmp_len);
-		name_len = name_len - (tmp_len + 2);
-		name_str = tmp + 2;
-		orig_obj = NULL;
-	} else if (Z_TYPE_P(classname) == IS_OBJECT) {
-		orig_obj = classname;
-	} else {
-		orig_obj = NULL;
+	switch (Z_TYPE_P(classname)) {
+		case IS_STRING:
+			orig_obj = NULL;
+
+			if (ZEND_NUM_ARGS() == 1) {	
```

Comment:
```suggestion
			if (!name_str) {	
```

But we'll need @cmb69 's approval for making the second parameter nullable for PHP 7.3. But it should absolutely work for PHP 8.

Reply:
Not a strict no from me, but I'd rather not change the signature in a stable version. @derickr what do you think about that?

Reply:
Yeah, Probably adding the `!` modifier only makes sense for PHP 8.

Reply:
Maybe to not complicate it further it would be better if I revert the code back to the first commit in this PR, and then apply [this feedback](https://github.com/php/php-src/pull/5995#pullrequestreview-468075204). Then, for PHP 8, a different solution like having a nullable 2nd parameter, could be applied. 

However, I am not sure if this is desirable and, in case it is, about the procedure to make a change in different PHP versions.

What do you think?

Reply:
I think it's also a sensible idea. We could also ask from ourselves if this is really needed to be fixed in PHP 7 at all? Personally, I'd only fix it in 8, but ultimately, it's Christoph's and Derick's call :)

Reply:
I think that fixing the erroneous error message for PHP-7.3 makes sense (given there is no change in the signature, i.e. second parameter is not nullable). After that has been merged into PHP-7.4 and the "master" branch, another PR removing the function overload (i.e. second parameter made nullable) could follow up.